test.cpp

```diff
@@ -346,7 +369,7 @@ int main()
 	test5();
 	test6();
 	test7(); // summarized results
-	test8(); // collect_scope
+	(new Test8())->run();
```

Comment:
Surely there is a nicer way to do this that doesn't also leak memory? Some tools will not like that.

Reply:
Thanks, I've updated it to use smart pointers.